File: src/main/kotlin/com/github/sydowma/fastplugin/settings/SettingsConfigurable.java
package com.github.sydowma.fastplugin.settings;


import com.intellij.openapi.options.Configurable;
import com.intellij.openapi.options.ConfigurationException;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class SettingsConfigurable implements Configurable {
    private SettingsComponent settingsComponent;

    @Nls(capitalization = Nls.Capitalization.Title)
    @Override
    public String getDisplayName() {
        return "Quick Prompt Settings";
    }

    @Nullable
    @Override
    public JComponent createComponent() {
        settingsComponent = new SettingsComponent();
        return settingsComponent.getPanel();
    }

    @Override
    public boolean isModified() {
        SettingsState settings = SettingsState.getInstance();
        return settingsComponent.isModified(settings);
    }

    @Override
    public void apply() throws ConfigurationException {
        SettingsState settings = SettingsState.getInstance();
        settingsComponent.apply(settings);
    }

    @Override
    public void reset() {
        SettingsState settings = SettingsState.getInstance();
        settingsComponent.reset(settings);
    }

    @Override
    public void disposeUIResources() {
        settingsComponent = null;
    }
}

File: src/main/kotlin/com/github/sydowma/fastplugin/settings/SettingsConfigurable.kt
package com.github.sydowma.fastplugin.settings

import com.github.sydowma.fastplugin.settings.SettingsState.Companion.instance
import com.intellij.openapi.options.Configurable
import com.intellij.openapi.options.ConfigurationException
import org.jetbrains.annotations.Nls
import javax.swing.JComponent


class SettingsConfigurable : Configurable {
    private var settingsComponent: SettingsComponent? = null

    override fun getDisplayName(): @Nls(capitalization = Nls.Capitalization.Title) String {
        return "Quick Prompt Settings"
    }

    override fun createComponent(): JComponent? {
        settingsComponent = SettingsComponent()
        return settingsComponent!!.panel
    }

    override fun isModified(): Boolean {
        val settings = instance
        return settingsComponent!!.isModified(settings)
    }

    @Throws(ConfigurationException::class)
    override fun apply() {
        val settings = instance
        settingsComponent!!.apply(settings)
    }

    override fun reset() {
        val settings = instance
        settingsComponent!!.reset(settings)
    }

    override fun disposeUIResources() {
        settingsComponent = null
    }
}
